clang/test/CodeGenHLSL/vk-features/vk.spec-constant.hlsl

```diff
@@ -0,0 +1,210 @@
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --check-globals all --include-generated-funcs --version 5
+// RUN: %clang_cc1 -finclude-default-header -x hlsl -triple \
+// RUN:   spirv-unknown-vulkan-compute %s -emit-llvm -disable-llvm-passes \
+// RUN:   -o - | FileCheck %s
+
+[[vk::constant_id(1)]]
+const bool bool_const = true;
```

Comment:
Shall we test case where the `constant_id` is used twice, even with different types?
DXC allows it, there is no validation error, and somehow the pipeline works fine when I tried on mesa (float & int sharing the same SpecID and setting it)